crates/miden-objects/src/note/note_tag.rs

```diff
@@ -181,41 +199,46 @@ impl NoteTag {
             return Err(NoteError::NoteTagUseCaseTooLarge(use_case_id));
         }
 
-        let execution_bits = LOCAL_EXECUTION_WITH_ALL_NOTE_TYPES_ALLOWED;
         let use_case_bits = (use_case_id as u32) << 16;
         let payload_bits = payload as u32;
 
-        Ok(Self(execution_bits | use_case_bits | payload_bits))
+        Ok(Self::LocalAccount(LOCAL_ACCOUNT | use_case_bits | payload_bits))
     }
 
     // PUBLIC ACCESSORS
     // --------------------------------------------------------------------------------------------
 
-    /// Returns true if the note is intended for execution by a specific account.
+    /// Returns true if the note is intended for execution by a specific account.(network or local)
     ///
-    /// A note is intended for execution by a single account if the first two bits are zeros
+    /// A note is intended for execution by a single account if the first two bits are `0b00` or
+    /// `0b10`
```

Comment:
I would remove this. Now that we have the enum, we can just refer to the variants instead (as suggested in a previous comment).